pdns/recursordist/pdns_recursor.cc

```diff
@@ -2027,14 +2027,19 @@ void getQNameAndSubnet(const std::string& question, DNSName* dnsname, uint16_t*
   const size_t headerSize = /* root */ 1 + sizeof(dnsrecordheader);
   const uint16_t arcount = ntohs(dhPointer->arcount);
 
-  for (uint16_t arpos = 0; arpos < arcount && questionLen > (pos + headerSize) && (lookForECS && !foundECS); arpos++) {
+  for (uint16_t arpos = 0; arpos < arcount && questionLen >= (pos + headerSize) && (lookForECS && !foundECS); arpos++) {
```

Comment:
Right now we are only executing this loop if we are looking for ECS, and we know there is not going to be an ECS option if `questionLen == (pos + headerSize)`. I guess the question is, do we want to make to always enable this parsing so we can get the EDNS version? If so we need to get rid of the `lookForECS` condition, and likely change the one on line 2259.

Reply:
If we're PB logging, `needECS` is set to true (2220) and `getQNameAndSubnet` is called with a set `ednssubnet`. I think I'll rename `needECS` to `needEDNSparse` and adapt `getQNameAndSubnet`.

Reply:
Ah, I was looking for something like that and I still missed it, thanks!